### PR TITLE
Feat: Pet Nurturing

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Nurturing` feature for enhanced metadata management.
	- Added `InteractCube` functionality for improved cube interactions.
	- Implemented methods for managing nurturing items and their states.

- **Bug Fixes**
	- Resolved MySQL connection issues in plot loading.

- **Improvements**
	- Enhanced logging during parsing processes.
	- Streamlined cube placement and interaction logic.
	- Improved error handling for cube interactions and plot management.
	- Updated gameplay mechanics for nurturing and cube interactions.

- **Documentation**
	- Updated handling of nurturing and cube interactions in gameplay mechanics.

### CODE REVIEW

File: Maple2.Database/Storage/Game/GameStorage.Nurturing.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                ClaimedGiftForStage = 1,
+                CreationTime = DateTime.Now,
+                LastFeedTime = null,
+                PetBy = [],
+            };
+            Context.Nurturing.Add(nurturing);
+            bool success = Context.TrySaveChanges();
+            if (!success) {
+                return null;
+            }
```
Comment (darkvergus) [inline Maple2.Database/Storage/Game/GameStorage.Nurturing.cs:33]: why not do if (!Context.TrySaveChanges()) return null;?

File: Maple2.Database/Storage/Game/GameStorage.Nurturing.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+            return new Maple2.Model.Game.Nurturing(result.Exp, result.ClaimedGiftForStage, result.PetBy, result.CreationTime, result.LastFeedTime, metadata.Nurturing);
+        }
+
+        public Maple2.Model.Game.Nurturing? CreateNurturing(long accountId, PlotCube plotCube) {
+            var nurturing = new Nurturing {
+                AccountId = accountId,
+                ItemId = plotCube.ItemId,
+                Exp = 0,
+                ClaimedGiftForStage = 1,
```
Comment (Zintixx) [inline Maple2.Database/Storage/Game/GameStorage.Nurturing.cs:24]: Default is 1?

File: Maple2.Database/Model/Map/Nurturing.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Maple2.Database.Model;
+
+internal class Nurturing {
+    public long AccountId { get; set; }
+    public int ItemId { get; set; }
+    public long Exp { get; set; }
+    public short ClaimedGiftForStage { get; set; }
+    public DateTime CreationTime { get; set; }
+    public DateTime? LastFeedTime { get; set; }
```
Comment (Zintixx) [inline Maple2.Database/Model/Map/Nurturing.cs:13]: I don't think this needs to be null. We can just ensure the conversion is done properly.

File: Maple2.Server.Game/Packets/FunctionCubePacket.cs [hunk 4]
```diff
@@ -13,54 +15,62 @@ private enum Command : byte {
...
         return pWriter;
     }
 
     public static ByteWriter UseFurniture(long characterId, PlotCube cube) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.Furniture);
         pWriter.WriteLong(characterId);
-        pWriter.WriteUnicodeString(cube.InteractId);
-        pWriter.WriteBool(cube.InteractState is InteractCubeState.InUse);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
```
Comment (Zintixx) [inline Maple2.Server.Game/Packets/FunctionCubePacket.cs:42]: Do we need to surpress this? Why not pass in InteractCube instead?

File: Maple2.Server.Game/Packets/FunctionCubePacket.cs [hunk 5]
```diff
@@ -13,54 +15,62 @@ private enum Command : byte {
...
         pWriter.WriteLong(DateTimeOffset.UtcNow.ToUnixTimeSeconds());
         return pWriter;
     }
+
+    public static ByteWriter Feed(long itemUid, PlotCube cube) {
+        var pWriter = Packet.Of(SendOp.FunctionCube);
+        pWriter.Write<Command>(Command.Feed);
+        pWriter.WriteLong(itemUid);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
```
Comment (Zintixx) [inline Maple2.Server.Game/Packets/FunctionCubePacket.cs:70]: ..and as opposed to here, where we need to have the cube ID and all the InteractCube stuff. I feel like we should either just:
1. Make InteractCube non nullable and set a default (empty) InteractCube on all
2. Pass in the cube ID and InteractCube instead

**Review by Zintixx — CHANGES_REQUESTED** [review]

File: Maple2.Database/Storage/Game/GameStorage.Nurturing.cs (hunk 2 above)
Comment (AngeloTadeucci) [reply Maple2.Database/Storage/Game/GameStorage.Nurturing.cs:24]: yes

**Review by Zintixx — APPROVED** [review]

**Pull request merged**